### PR TITLE
(enhancement #200) Mother hen updates

### PR DESCRIPTION
**Pull request by kepler471**
New spawn type, 'Chicks'

Updated visuals to use the spawner sprites David made.

Need some help for this, having a little trouble with the collisions of the spawns.

Also would like to know how we feel like using this for the Boss rooms, and in that case we need to decide how we want it to behave.

Last point: I have used the MotherHen and adapted it, if we also want the mother hen in the game I can just move these changes to a new baddie, however I thought as she was a static baddie anyway, she was well suited for this Spawner.

spawner2

### CODE REVIEW

**Comment by whitew1994WW** [comment]
Will review the code tomorrow but the principle sounds good. If its a static boss node we could have it spawn different baddie types depending on the level. Lvl 1 - bear bois and flybois, lvl2 - prev + tanky bois , lvl3 - prev + bombers? lvl4 - prev + shooty bois, lvl 5 ???

**Comment by whitew1994WW** [comment]
Can we have its size scale in time with the flashing too? Should be simple to do with an animation player synced up

**Comment by kepler471** [comment]
> 
> 
> Will review the code tomorrow but the principle sounds good. If its a static boss node we could have it spawn different baddie types depending on the level. Lvl 1 - bear bois and flybois, lvl2 - prev + tanky bois , lvl3 - prev + bombers? lvl4 - prev + shooty bois, lvl 5 ???

Ye i like that idea, I will take a look at how we can do that. James has already added some nice code to handle the spawning (**BaddieLaser/BaddieSpawner.gd**), so can build from that.

**Comment by kepler471** [comment]
> 
> 
> Can we have its size scale in time with the flashing too? Should be simple to do with an animation player synced up

Like it's pulsating? Ye can try that too

**Comment by kepler471** [comment]
#221 should fix my problems with collisions!!

**Comment by whitew1994WW** [comment]
I put it into the splash screen to test and shot at it. Looks really cool with all of the baddies spawning as you shoot it. Ithink they should spawn on regular intervals as well so that you can't hang around too long. Also im getting this error when it dies: 
image

File: Moonshot/baddies/characters/MotherHen.gd [hunk 1]
```diff
@@ -32,4 +32,4 @@ func on_hit(instance_id, damage) -> void:
 		attributes._on_hit(damage, global_position)
 
 func on_end() -> void:
-	call_deferred("free")
+	attributes.set_properties({"should_damage_on_collision": false})
```
Comment (whitew1994WW) [inline Moonshot/baddies/characters/MotherHen.gd:35]: How come, do we want to be able to stand on top of it?

**Review by whitew1994WW — CHANGES_REQUESTED** [review]
Apart from the bug mentioned above looks good to me. How do you set the baddie type?

File: Moonshot/baddies/characters/MotherHen.gd (hunk 1 above)
Comment (kepler471) [reply Moonshot/baddies/characters/MotherHen.gd:35]: Yes, want to be able to stand on it when dead, and then we can maybe inject it with the syringe. And when that happens, maybe it then destroys, and reveals the elevator shaft you you spoke about with Bill?

**Comment by kepler471** [comment]
I'm gonna add these 

> 
> 
> I put it into the splash screen to test and shot at it. Looks really cool with all of the baddies spawning as you shoot it. Ithink they should spawn on regular intervals as well so that you can't hang around too long. Also im getting this error when it dies:
> image

Bug fixed

File: Moonshot/player/assets/sounds/running.wav.import [hunk 2]
```diff
@@ -0,0 +1,21 @@
+[remap]
+
+importer="wav"
+type="AudioStreamSample"
+path="res://.import/running.wav-28243bd05bb5595ac0ae7202626ad925.sample"
```
Comment (wattsit) [inline Moonshot/player/assets/sounds/running.wav.import:5]: What's with all the unrelated files in this pull?

**Review by wattsit — CHANGES_REQUESTED** [review]

**Comment by kepler471** [comment]
Ok, so this now contains a tooling change to all enemies, I have attemped to make it possible to change their direction by adding a button to the inspector. There seem to be some issues with this working but want to get this branch in.

I will add new mechanics to hen in a new branch I think.

File: Moonshot/baddies/BaddieLaser/BaddieSpawner.gd [hunk 3]
```diff
@@ -1,31 +1,34 @@
...
 	spawn()
-	var timer: Timer = Timer.new()
-	add_child(timer)
-	timer.start(0.5)
+
+func set_direction(x: float) -> void:
+	spawn_dir = sign(x)
 
 func spawn():
+	print("Spawned")
```
Comment (Jamesbarford) [inline Moonshot/baddies/BaddieLaser/BaddieSpawner.gd:20]: remove

File: Moonshot/baddies/characters/Chick.gd [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+		"inital_hp": 1.0,
+		"gravity": 50,
+		"damage_to_player": 0.02,
+		"floor_vector": Vector2(0, -1),
+		"should_damage_on_collision": true
+	})
+
+func _ready():
+	attributes.set_sprite($AnimatedSprite)
+	print("chicks mdir ::: ", mother_dir)
```
Comment (Jamesbarford) [inline Moonshot/baddies/characters/Chick.gd:28]: remove

**Review by Jamesbarford — CHANGES_REQUESTED** [review]
